mqtt/mqtt-broker/src/translation.rs

```diff
@@ -0,0 +1,81 @@
+use crate::ClientEvent;
+use lazy_static::lazy_static;
+use mqtt3::proto;
+use regex::Regex;
+
+pub fn translate_incoming(client_id: &str, event: ClientEvent) -> ClientEvent {
+    match event {
+        ClientEvent::Subscribe(s) => ClientEvent::Subscribe(subscribe(client_id, s)),
+        ClientEvent::Unsubscribe(u) => ClientEvent::Unsubscribe(unsubscribe(client_id, u)),
+        ClientEvent::PublishFrom(p) => ClientEvent::PublishFrom(publish_from(client_id, p)),
+        e => e,
+    }
+}
+
+pub fn translate_outgoing(packet: proto::Packet) -> proto::Packet {
+    match packet {
+        proto::Packet::Publish(p) => proto::Packet::Publish(publish_to(p)),
+        p => p,
+    }
+}
+
+fn subscribe(client_id: &str, subscribe: proto::Subscribe) -> proto::Subscribe {
```

Comment:
I'd say we need to introduce a notion of IothubTopicFilter here, so we can get rid of cloning topics and re-use elsewhere
```rust
fn subscribe(client_id: ClientId, subscribe: proto::Subscribe) -> proto::Subscribe {
    let subscribe_to = subscribe.subscribe_to.into_iter().map(|to| {
        proto::SubscribeTo {
            topic_filter: IotHubTopic::new(client_id.clone(), topic_filter).to_string(),
            qos: to.qos
        }
    }.collect();

    proto::Subscribe {
        packet_identifier:subscribe.packet_identifier,
        subscribe_to,
    }
}

struct IotHubTopic(String);

impl IotHubTopic {
    fn new(client_id:ClientId, topic_filter: String) -> Self {
        let topic = match &topic_filter {
            "$iothub/twin/res" => format!("$edgehub/{}/twin/res", client_id),
            "$iothub/twin/GET" => format!("$edgehub/{}/twin/get", client_id),
            _ => topic,
        };
    
        Self(topic)
    }
}

impl ToString for IotHubTopic {
    fn to_string(self) -> String {
        self.0
    }
}
```